### ISSUE TITLE
Migrate the backend

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#84]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 10:36 GMT+2:***

As a shepard developer, I want that the full functionality of shepard is available in the new quarkus backend in order to be able to focus all development in the new backend and archive the repository for the old backend. 

ACs: 

* ~~the backend image is now deploying the new quarkus backend~~ See decision 30.07. in the comments
* the quarkend image is ready to replace the backend image
* pipelines for quarkend are available and successful
* Automated tests are running and cover the full functionality
* ~~there is no visible difference between the two images if you see the backend as a black box~~
* ~~end user can use the image without any changes needed~~ (Admins will have to change some things)
* Add the findings on potential improvements in the backlog

Out of Scope: 

* actual implementation of additional framework features or improvements 

Hints / Work packages: 

* how does automated test work with quarkus 
** setup test infrastructure before we can actually migrate tests 
* New backend offers the same REST-API 
** how does it look like with Quarkus?
* Filters and Interceptors / Authentication works as before
** security and auth 
* New backend connects with all three databases. Existing shepard installations can switch the backend without needing to touch the databases.
** Neo4j + organizational services and DAO 
** InfluxDB connector + timeseries service 
** mongoDB connector + file services + structured data service
* Integration tests run again 
* Merging strategy / Alignment with Roland on parallel versioning feature development 
** initial plan is to merge Rolands feature after complete development 
** do an appointment to align the merging strategy & then follow up

**Assignees:** xit-paul-reinhardt

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/work_items/84*

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 10:40 GMT+2:***

added #86 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 10:41 GMT+2:***

added #87 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 10:42 GMT+2:***

added #88 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 10:42 GMT+2:***

added #89 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 10:44 GMT+2:***

added to epic &5

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 10:50 GMT+2:***

Decision 30.07.: There shall be one last backend release in the beginning of August. That’s why the migration will happen in a seperate task as soon as the release and this story are done. I will adjust the ACs accordingly.

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 10:52 GMT+2:***

added #90 as child task

**Comment by noheton** [comment]
***In GitLab by @thaase on Jul 31, 2024, 11:00 GMT+2:***

marked this issue as blocking #73

**Comment by noheton** [comment]
***In GitLab by @thaase on Jul 31, 2024, 11:41 GMT+2:***

marked this issue as blocking #94

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Jul 31, 2024, 16:24 GMT+2:***

added #104 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 1, 2024, 14:11 GMT+2:***

Current State of ACs:

* [x] ~~the backend image is now deploying the new quarkus backend~~ See decision 30.07. in the comments 
* [x] the quarkend image is ready to replace the backend image
* [x] pipelines for quarkend are available and successful
* [x] Automated tests are running and cover the full functionality
* [x] ~~there is no visible difference between the two images if you see the backend as a black box~~
* [x] ~~end user can use the image without any changes needed~~ (Admins will have to change some things)
* [x] Add the findings on potential improvements in the backlog

For Testing:

* Setup the quarkend in VSCode or IntelliJ following quarkend/README.md
* Run the docker-compose setup in quarkend
* Run unit & integration tests
* Open the frontend locally and check if everything works
* Open the frontend of the Xitaso dev instance at https://frontend.desh.dev.xitaso.com/ and check if everything works (it uses the quarkend already)
* Open Swagger at https://quarkend.desh.dev.xitaso.com/shepard/doc/swagger-ui and check if everything works

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 1, 2024, 14:12 GMT+2:***

added #108 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 1, 2024, 14:13 GMT+2:***

added #109 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 2, 2024, 09:53 GMT+2:***

added #110 as child task

**Comment by noheton** [comment]
***In GitLab by @thaase on Aug 2, 2024, 12:00 GMT+2:***

added #111 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 2, 2024, 15:27 GMT+2:***

mentioned in commit eb091494cd1206eb9f3956431a8e7a05af8d6aad

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 2, 2024, 15:30 GMT+2:***

mentioned in commit e7547a145aedb2ef6ea67cd7cc723bbbe627994d

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 2, 2024, 17:47 GMT+2:***

mentioned in commit 6fd80c482694834d3be2144caada515352cbcd77

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 5, 2024, 08:31 GMT+2:***

Decision 01.08.: Since the native image does not work right now, we stick to a JVM image temporarily. The switch to a native image is extracted to #106

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 5, 2024, 13:10 GMT+2:***

mentioned in commit 146dde9085d2e3d59b63cc689700e6ca2f61b5fb

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Aug 5, 2024, 17:01 GMT+2:***

I tested out the individual steps in a new environment and almost everything works as expected.
However, I noticed the followings things:
- the README.md is missing a step to start the docker-compose.yml (readers could think that the vscode launch config is configured to start all needed containers, which is not the case)
- the README.md does not mention to set the VUE_APP_OIDC_AUTHORITY, and VUE_APP_CLIENT_ID in the .env file. These ENVs are used in the docker-compose file
- I was not able to upload files larger than 10MB, neither in the swagger-ui nor in the frontend. I think this is probably related to a quarkus issue limiting the max POST payload size.

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 7, 2024, 08:32 GMT+2:***

added #128 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 7, 2024, 08:42 GMT+2:***

mentioned in commit 9c51f51505c0645c6f1b43ae7b3c3f799cb98efc

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 7, 2024, 08:44 GMT+2:***

I fixed the readme issues in https://gitlab.com/dlr-shepard/shepard/-/merge_requests/163 and https://gitlab.com/dlr-shepard/shepard/-/merge_requests/164. I created a subtask for the file upload size #128.

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 7, 2024, 13:57 GMT+2:***

mentioned in commit 7a53f288301b53d07433da9bb7dfd180e8df3b72

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 7, 2024, 14:11 GMT+2:***

mentioned in commit 59c2367d7a51ad8f2e2c8e0f5f54df551dafd45e

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 8, 2024, 09:36 GMT+2:***

mentioned in commit cebeba0396067e76ebbc9297e6ac7f9d8f56ffa4

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 8, 2024, 10:05 GMT+2:***

mentioned in commit 388d0a847d14fa3c360ddf46b63554cc7283265b

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 8, 2024, 10:05 GMT+2:***

mentioned in commit 49745dd78491698c209bdb5a063fe7b0f16ada05

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 8, 2024, 14:15 GMT+2:***

added #136 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 9, 2024, 15:06 GMT+2:***

added #142 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 12, 2024, 11:03 GMT+2:***

All issues that @xit-max-hey found during testing are resolved. It would be great if somebody could retest those so that we can close the story. His findings where (taken from https://gitlab.com/dlr-shepard/shepard/-/issues/84#note_2031724689): 

* the README.md is missing a step to start the docker-compose.yml (readers could think that the vscode launch config is configured to start all needed containers, which is not the case)
* the README.md does not mention to set the VUE_APP_OIDC_AUTHORITY, and VUE_APP_CLIENT_ID in the .env file. These ENVs are used in the docker-compose file
* I was not able to upload files larger than 10MB, neither in the swagger-ui nor in the frontend. I think this is probably related to a quarkus issue limiting the max POST payload size.

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Aug 12, 2024, 11:51 GMT+2:***

Testing:
- Installation in backend/README.md is missing setting up the `.env` file, which causes issues when running the project

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 12, 2024, 13:04 GMT+2:***

That's described in CONTRIBUTING.md.

**Comment by noheton** [comment]
***In GitLab by @thaase on Aug 12, 2024, 13:29 GMT+2:***

maybe an issue: https://gitlab.com/dlr-shepard/shepard/-/work_items/87#note_2040888383

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 12, 2024, 13:48 GMT+2:***

added #145 as child task

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Aug 12, 2024, 14:54 GMT+2:***

- Dev setup is working :white_check_mark:
- README.md now states a step to start the docker-compose.yml :white_check_mark:
- VUE_APP_OIDC_AUTHORITY and VUE_APP_CLIENT_ID are automatically set when adding the \`.env\` file :white_check_mark:
- Files over 10mb in size could be uploaded :white_check_mark:
- Tests are green :white_check_mark:
- Frontend available :white_check_mark:
- Swagger UI available :white_check_mark:

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 13, 2024, 11:37 GMT+2:***

Fixed in #145
